hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/StateNetworkInfo.java

```diff
@@ -103,16 +134,24 @@ public void updateFrom(@NonNull final State state) {
      */
     private Map<Long, NodeInfo> buildNodeInfoMap(final State state) {
         final var nodeInfos = new LinkedHashMap<Long, NodeInfo>();
-        final var rosterEntries = retrieve(state).rosterEntries();
+        final var rosterEntries = activeRoster.rosterEntries();
         final ReadableKVState<EntityNumber, Node> nodeState =
                 state.getReadableStates(AddressBookService.NAME).get(NODES_KEY);
         for (final var rosterEntry : rosterEntries) {
-            final var node = nodeState.get(
-                    EntityNumber.newBuilder().number(rosterEntry.nodeId()).build());
-            if (node != null) {
-                nodeInfos.put(rosterEntry.nodeId(), fromRosterEntry(rosterEntry, node));
-            }
+            // At genesis the node store is derived from the roster, hence must have info for every
+            // node id; and from then on, the roster is derived from the node store, and hence the
+            // node store must have every node id in the roster.
+            final var node = requireNonNull(nodeState.get(new EntityNumber(rosterEntry.nodeId())));
+            // Notice it's possible the node could be deleted here, because a DAB transaction removed
+            // it from the future address book; that doesn't mean we should stop using it in the current
+            // version of the software
```

Comment:
Fascinating, we should write a test that does this! (in a future PR)

Reply:
This is already covered in DAB upgrade tests. But we can add specific TSS tests in future PRs also.